lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -6,11 +6,12 @@
 
 module MiqAeEngine
   DEFAULT_ATTRIBUTES = %w( User::user MiqServer::miq_server object_name )
+  V2_PASSWORD_REGEXP = /v([0-9]+)%3A%7B(.*?)%7D/ # for URI encoded string "v2:{.*}"
 
   def self.instantiate(uri, user)
-    $miq_ae_logger.info("MiqAeEngine: Instantiating Workspace for URI=#{uri}")
+    $miq_ae_logger.info("MiqAeEngine: Instantiating Workspace for URI=#{sanitize_encoded_string(uri)}")
```

Comment:
@lfu Why can't we use MiqPassword.sanitize_string here?

Reply:
The string is URL encoded so it no longer matches the regex in MiqPassword.

Reply:
@lfu @gmcculloug 
Are we getting passwords as URL's?
Can the function sanitize_encoded_string to MiqPassword so its all in one place and we can control if it should be asterisk or [FILTERED] or something else

Reply:
Its the password coming in from Dialogs any dialog_field that is marked as encrypted comes in with the v2 string.